### PR TITLE
*: Refine error message when deadline exceeded

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6233

Problem Summary:

### What is changed and how it works?

* Refine error message when establish disagg task timeout
* Refine error logging when wn handling expired EstablishDisaggTask request
* Refine error logging when query fails with the keyspace id of table
* Simplify some logic in learner read

Before this PR
```sql
-- EstablishDisaggTask timeout
> select count(*) from sbtest1;
(1105, 'other error for mpp stream: Code: 0, e.displayText() = DB::Exception: EstablishDisaggTask Failed4: Deadline Exceeded, e.what() = DB::Exception,')
-- Wait index timeout
> select count(*) from sbtest1;
(1105, 'other error for mpp stream: Code: 0, e.displayText() = DB::Exception: EstablishDisaggregatedTask failed: Region unavailable, region_id=199035950 wait_index=1528819 applied_index=1528819: (while doing learner read for table,  logical_table_id=422), addr=10.2.12.81:9520, e.what() = DB::Exception,')
```

After this PR
```sql
-- EstablishDisaggTask timeout
> select count(*) from sbtest1;
(1105, 'other error for mpp stream: Code: 159, e.displayText() = DB::Exception: EstablishDisaggregated execution was interrupted, maximum execution time exceeded, wn_address=10.2.12.81:9520 MPP<gather_id:<gather_id:1, query_ts:1697120494131440477, local_query_id:39, server_id:3861858, start_ts:444889954811904001, resource_group: >,task_id:1>, e.what() = DB::Exception,')
-- Wait index timeout
> select count(*) from sbtest1;
(1105, 'other error for mpp stream: Code: 0, e.displayText() = DB::Exception: EstablishDisaggregatedTask failed: Region unavailable, region_id=199035950 wait_index=1528819 applied_index=1528819: (while doing learner read for table, keyspace=4294967295 logical_table_id=422), addr=10.2.12.81:9520, e.what() = DB::Exception,')
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/KVStore/Read/LearnerRead.cpp [hunk 1]
```diff
@@ -356,23 +359,23 @@ LearnerReadSnapshot doLearnerRead(
 
     const size_t num_regions = regions_info.size();
 
-    // TODO: refactor this enormous lambda into smaller parts
+    const auto start_time = Clock::now();
     UnavailableRegions unavailable_regions;
     LearnerReadStatistics stats;
-    const auto batch_wait_index = [&](const size_t region_begin_idx) -> void {
+
+    // TODO: refactor this enormous lambda into smaller parts
```
Comment (CalvinNeo) [inline dbms/src/Storages/KVStore/Read/LearnerRead.cpp:366]: the `batch_wait_index` is removed, so this is not a todo any more?

Comment (JaySon-Huang) [reply dbms/src/Storages/KVStore/Read/LearnerRead.cpp:366]: The code block of read index && wait index still takes about 150 lines, I think we could break it into smaller parts (and add some unit testing) later

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Comment by JaySon-Huang** [comment]
/hold
seems some unit test hangs

**Pull request merged**